### ISSUE TITLE
Incosistent latex output with doubled single quotes

### ISSUE THREAD

**Issue by marcin-serwin** [jgm/pandoc#7332]
## Current behaviour

When converting this markdown file
```
``text''
```
directly to pdf (`pandoc example.py -o example.pdf`) the single quotes are treated as latex double quote and look nice, however converting to tex (`pandoc example.py -o example.tex`) produces the following output
```
``text'\,'
```
which makes the latex treat single quotes as separate.

## Expected behaviour
`pandoc example.py -o example.tex` should produce
```
``text''
```

## pandoc -v
```
pandoc 2.13
Compiled with pandoc-types 1.22, texmath 0.12.3, skylighting 0.10.5.1,
citeproc 0.3.0.9, ipynb 0.1.0.1
```

**Comment by jgm** [comment]
That's right.  In TeX ` `` ` means "left double quote."  But in Markdown it doesn't.

**Comment by marcin-serwin** [comment]
I was talking about "right double quote" (or `''` in latex), and the issue is that it's inconsistent compared to direct pdf output. I'd expect the commands
```
pandoc example.py -o example.pdf
```
and
```
pandoc -s example.py -o example.tex ; pdflatex example.tex
```
to produce similar documents but the former produces this:
![image](https://user-images.githubusercontent.com/12128106/120111657-557ad080-c162-11eb-9ca0-1ea8e0c4aba3.png)
while the latter produces this:
![image](https://user-images.githubusercontent.com/12128106/120111680-69263700-c162-11eb-8237-b604b3218739.png)

**Comment by marcin-serwin** [comment]
So if `''` is not "right double quote in markdown shouldn't this be also reflected when converting directly to pdf?

**Comment by jgm** [comment]
OK, I see the issue.  The discrepancy is explained by the fact that the pdf-via-latex path turns off the "smart" option and just uses unicode quote characters in the tex intermediary.  So the tex intermediate file has
```
``text’’
```
Note that the markdown `` ` `` characters are left unchanged, since this is not a way of writing a left quote in markdown, while the `` ' `` characters are turned to right curly quotes.)  When this is processed by latex, the ` `` ` gets turned into a double left curly quote, and the ` ’’ ` into two consecutive single right curly quotes.  Visually this is impossible to distinguish from a right double quote.

When `smart` is enabled for LaTeX output, we use the ligatures, and we insert the thin space to prevent the two consecutive ` ' ` characters from creating a right double quote.

There are a few things we might consider doing here to make things a bit better.

- [ ] Escaping `` ` `` characters so they render literally instead of being treated as ligatures.  We currently do this only in code contexts, but we should do it everywhere.
- [ ] inserting the thin space between consecutive unicode quote characters

**Comment by marcin-serwin** [comment]
Is there a way to disable `smart` just for quotes? My usecase is that I need to produce lower double quotes like this
![image](https://user-images.githubusercontent.com/12128106/120113309-6418b600-c169-11eb-9c34-f44805d7f355.png)
or
```
,,text''
```
in latex, but turning off `smart` completely will in turn break my usage of en-dashes and em-dashes (`--` and `---`).

**Comment by jgm** [comment]
There isn't a way to do this built into pandoc.  But you may be able to do it by inserting some tex commands in the header?  Google about disabling tex ligatures.
